### PR TITLE
TagAttribute

### PR DESCRIPTION
**Pull request by lukaszlenart**
WIP

### CODE REVIEW

File: core/src/main/java/org/apache/struts2/views/TagAttribute.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    public String stripedExpression() {
+        if (isExpression()) {
+            return value.substring(2, value.length() - 1);
+        } else {
+            return value;
+        }
+    }
+
+    public TagAttribute escaped(){
```
Comment (JCgH4164838Gh792C124B5) [inline core/src/main/java/org/apache/struts2/views/TagAttribute.java:54]: Minor suggestion (spacing/readability):  change `(){`  to `() {`

**Comment by JCgH4164838Gh792C124B5** [comment]
Hello @lukaszlenart .

This looks like an interesting change.   😄  
I will try to review it in more detail when I have a chance.  It looks like the change might be introducing a failure (java.lang.ClassCastException) in the Showcase tests, as it stands right now (if I read the Travis CI reports correctly).

**Review by JCgH4164838Gh792C124B5 — CHANGES_REQUESTED** [review]
Hi @lukaszlenart .

I downloaded the branch for these changes, took another look, and attempted to build things locally.

Adding the additional methods suggested in the review and then modifying the components (either using the new conversion method, or relying on the implicit `toString()` addition) seemed to work for me locally.  This allowed the Showcase test that was failing to pass, and interactive tests for the Showcase UI components seemed to work OK.

I do not know if the changes suggested in this review are a good strategy or not, but I think in order to implement the `TagAttribute` there will need to be changes in places where the code assumes a component parameter will be a `TagAttribute` or `String` (so that it can handle either).

Depending on how many different types of `Component` parameter values might be encountered, some sort of general-purpose adapter might be a consideration (to convert from the various parameter types to `String` and back).  It does not look like it would be worth the overhead at the moment, though.

It looks like the idea for your PR works, with just a few boundary cases to deal with first.  Good work.  👍

**Comment by lukaszlenart** [comment]
I'm closing this one - there are some tags that makes implementing this quite hard.

**Pull request closed without merging**